### ISSUE TITLE
Arcmining vent waves spawn inside rocks

### ISSUE THREAD

**Issue by Sadboysuss** [tgstation/tgstation#82249]
## Issue Summary
Activating an arcmining vent destroys some rocks around it, but doesn't prevent the agressive mobs from spawning inside the rocks.

## Round ID:
N/A

## Reproduction:
1. Find an ore vent
2. Don't clear the area of rocks
3. Start the extraction process
4. Watch in horror as legion skulls appear but you can't find the legion anywhere.

**Comment by ArcaneMusic** [comment]
So, the bug should be localized to the spawner component from what I understand, so it might be slightly more widespread than described (but, in all likelyhood, it'll be just ore vents lets be real)

within `/datum/component/spawner/proc/try_spawn_mob()`,

```
for(var/i in 1 to adjusted_spawn_count)
		var/atom/created
		var/turf/picked_spot //Follow this part

		if(spawn_distance == 1)
			created = new chosen_mob_type(spawner.loc)
		else if(spawn_distance >= 1 && spawn_distance_exclude >= 1)
			picked_spot = pick(turf_peel(spawn_distance, spawn_distance_exclude, spawner.loc, view_based = TRUE)) //This is what we should be using for the vent, since we don't want to spawn anything next to the vent
			if(!picked_spot)
				picked_spot = pick(circle_range_turfs(spawner.loc, spawn_distance)) //And in theory, this should be the default if there's no valid turfs in view range based on `turf_peel()`
			created = new chosen_mob_type(picked_spot)
		else if (spawn_distance >= 1)
			picked_spot = pick(circle_range_turfs(spawner.loc, spawn_distance))
			created = new chosen_mob_type(picked_spot)
```
Looks like turf_peel is picking lavaland rock tiles since they're considered solid turfs, but they are totally in o_view based on the fact you'll still "see" the first rock wall in a radius

A good fix for this should likely also be a fix for when someone walls off an ore vent to block off all enemy spawns

**Comment by ArcaneMusic** [comment]
Fixed this with #83295.
